src/plugins/syscalls/syscalls.cpp

```diff
@@ -194,6 +199,38 @@ static event_response_t win_cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
         }
     }
 
+    if(!wrapper->ret_trap_set && wsc->ret != VOID){
```

Comment:
This is still problematic. Ideally we would create a new trap for each return which then would get released in their callback. That would ensure that the correct return address is trapped for each syscall.